spec/policy/url_rewriting/url_rewriting_spec.lua

```diff
@@ -78,5 +81,135 @@ describe('URL rewriting policy', function()
 
       assert.stub(ngx.req.set_uri).was_called_with('/some_path/new/123/new')
     end)
+
+    it('can add arguments to the query', function()
+      local config_to_add_args = {
+        query_args_commands = {
+          { op = 'add', arg = 'an_arg', value = 'a_val' }
+        }
+      }
+      local url_rewriting = URLRewriting.new(config_to_add_args)
+
+      url_rewriting:rewrite()
+
+      assert.stub(ngx.req.set_uri_args).was_called_with({ an_arg = 'a_val' })
+    end)
+
+    it('can replace arguments from the query', function()
```

Comment:
These tests duplicate tests of QueryParams.

I think we are over testing this a bit. We have 3 tests for the same code.
Not sure what is the best path forward though.

Reply:
I see your point. In particular, some of these tests check that `ngx.req.set_uri_args` was called with the proper arguments, but that shouldn't be a concern of `url_rewriting`. It should be of `query_params`, and it's being already tested on its specs.

However, it's a bit tricky to test this properly as `QueryParams.new()` is called from `rewrite()` and it does not make sense to inject an instance of QueryParams there.

Reply:
Check the latest commit. It mocks QueryParams to avoid testing its internals.